### PR TITLE
[Bouffalolab] build bl602 example by dev board and support *.flash.py…

### PR DESCRIPTION
**Pull request by wy-hh**
**Problem** 

chip-bl602-lighting-example.flash.py can't download image for bl602

### CODE REVIEW

File: .github/workflows/examples-bouffalolab.yaml [hunk 1]
```diff
@@ -52,21 +108,22 @@ jobs:
...
           .environment/pigweed-venv/bin/python3 scripts/tools/memory/gh_sizes.py bl702 bl702 lighting-app \
-            out/bouffalolab-bl706-iot-dvk-light/chip-bl702-lighting-example.out /tmp/bloat_reports/
+            out/bouffalolab-xt-zb6-devkit-light/chip-bl702-lighting-example.out /tmp/bloat_reports/
 
       - name: Build example BL702 Lighting App with RPCs
         timeout-minutes: 30
         run: |
           ./scripts/run_in_build_env.sh \
-            "./scripts/build/build_examples.py --target bouffalolab-bl706-iot-dvk-light-rpc build"
+            "./scripts/build/build_examples.py --target bouffalolab-xt-zb6-devkit-light-115200-rpc build"
```
Comment (andy31415) [inline .github/workflows/examples-bouffalolab.yaml:124]: If this changes, then integrations/cloudbuild/build-all.yaml should probably also change.

File: scripts/build/build/targets.py [hunk 2]
```diff
@@ -466,6 +459,8 @@ def BuildBouffalolabTarget():
         TargetPart('light', app=BouffalolabApp.LIGHT),
     ])
 
+    target.AppendModifier('shell', enable_shell=True)
+    target.AppendModifier('115200', baudrate="115200")
```
Comment (andy31415) [inline scripts/build/build/targets.py:463]: why is boudrate passed on as a string? It is a number right?

File: scripts/build/builders/bouffalolab.py [hunk 3]
```diff
@@ -42,11 +42,23 @@ def FlashBundleName(self):
 
 
 class BouffalolabBoard(Enum):
-    BL706_IoT_DVK = 1
-    BL706_NIGHT_LIGHT = 2
+    BL602_IoT_Matter_V1 = 1
```
Comment (andy31415) [inline scripts/build/builders/bouffalolab.py:45]: use enum.auto()

File: scripts/build/builders/bouffalolab.py [hunk 4]
```diff
@@ -42,11 +42,23 @@ def FlashBundleName(self):
...
+    BL602_IOT_DVK_3S = 2
+    BL602_NIGHT_LIGHT = 3
+    XT_ZB6_DevKit = 101
```
Comment (andy31415) [inline scripts/build/builders/bouffalolab.py:48]: There is no reason for this kind of large numeric jump. Just use enum.auto as we do not persist these as numeric.

File: scripts/build/builders/bouffalolab.py [hunk 5]
```diff
@@ -63,6 +75,8 @@ def __init__(self,
                  board: BouffalolabBoard = BouffalolabBoard.BL706_IoT_DVK,
                  enable_rpcs: bool = False,
                  module_type: str = "BL706C-22",
+                 baudrate: str = "2000000",
```
Comment (andy31415) [inline scripts/build/builders/bouffalolab.py:78]: why str?

File: .github/workflows/examples-bouffalolab.yaml (hunk 1 above)
Comment (wy-hh) [reply .github/workflows/examples-bouffalolab.yaml:124]: build-all.yaml updated

File: scripts/build/builders/bouffalolab.py (hunk 3 above)
Comment (wy-hh) [reply scripts/build/builders/bouffalolab.py]: changed

File: scripts/build/builders/bouffalolab.py (hunk 4 above)
Comment (wy-hh) [reply scripts/build/builders/bouffalolab.py]: changed

File: scripts/build/builders/bouffalolab.py (hunk 5 above)
Comment (wy-hh) [reply scripts/build/builders/bouffalolab.py]: changed to numeric

**Review by andy31415 — APPROVED** [review]

**Pull request merged**